create_derivatives/routines.py

```diff
@@ -78,8 +79,30 @@ class ManifestMaker:
 
 
 class AWSUpload:
-    # TO DO: upload files and PDFs
-    pass
+
+    def __init__(self):
+        self.aws_client = AWSClient(*settings.AWS)
+
+    def run(self, replace):
+        uploaded_bags = []
+        for bag in Bag.objects.filter(process_status=Bag.MANIFESTS_CREATED):
+            pdf_dir = join(bag.bag_path, 'data', 'PDF')
+            jp2_dir = join(bag.bag_path, 'data', 'JP2')
+            manifest_dir = join(bag.bag_path, 'data', 'MANIFEST')
+            check_dir_exists(pdf_dir)
+            check_dir_exists(jp2_dir)
+            check_dir_exists(manifest_dir)
```

Comment:
I'm not sure if the manifest directory should exist before this service is run; is it expecting a previous service to create that directory?

Reply:
Yes, sorry. I was thinking that the manifest creation routine would make that directory in the same way the PDF and JP2 directories were made. Does that make sense?